src/sinks/clickhouse.rs

```diff
@@ -77,9 +77,10 @@ impl SinkConfig for ClickhouseConfig {
         let tls_settings = TlsSettings::from_options(&self.tls)?;
         let client = HttpClient::new(cx.resolver(), tls_settings)?;
 
-        let sink = BatchedHttpSink::new(
+        let sink = BatchedHttpSink::with_retry_logic(
             self.clone(),
             Buffer::new(batch.size, self.compression),
+            ClickhouseRetryLogic::default(),
```

Comment:
I think this can be simply `ClickhouseRetryLogic,`

I'm surprised the previous non-use of `ClickhouseRetryLogic` wasn't caught as dead code by the compiler.

Reply:
No `ClickhouseRetryLogic,` not work :( I also surprised that clippy did not show any error about not used struct! I'd expect no error if struct was `pub`, but `ClickhouseRetryLogic` is not `pub`.

Reply:
Oh, of course, because `ClickhouseRetryLogic` has an inner. In that case the auto derive makes more sense.

Reply:
So, do you think it's better rever last changes? xD

Reply:
Yes, I think the result is simpler that way.